### PR TITLE
[KGA-19] dev: use Cairo1Helpers class for 'call_contract' for future-proofness

### PR DESCRIPTION
**Pull request by enitrat**
Uses a call to the Cairo1Helpers class to access the "new" call_contract syscall, that will __in the future__ have the ability to gracefully handle failed contract calls.

NOTE: The change implemented in the following PR does not fix [KGA-19] (https://github.com/code-423n4/2024-09-kakarot-findings/issues/49) because the Starknet network does not allow handling failed calls yet.

Attention to reviewer: The change to the Cairo1Helpers class needs to be merged in kakarot-ssj first. Discussion can be made here

### CODE REVIEW

File: cairo_zero/kakarot/accounts/library.cairo [hunk 1]
```diff
@@ -374,6 +374,35 @@ namespace AccountContract {
...
+    // @notice Execute a starknet call.
+    // @dev Used when executing a Cairo Precompile. Used to preserve the caller address.
+    // @param to The address to call.
+    // @param function_selector The function selector to call.
+    // @param calldata_len The length of the calldata array.
+    // @param calldata The calldata for the call.
+    // @return retdata_len The length of the return data array.
+    // @return retdata The return data from the call.
+    // @return success 1 if the call was successful, 0 otherwise.
+    func execute_starknet_call{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
```
Comment (obatirou) [inline cairo_zero/kakarot/accounts/library.cairo:387]: no tests ?

Comment (enitrat) [reply cairo_zero/kakarot/accounts/library.cairo:387]: All existing tests using Cairo Precompiles are testing this

**Comment by obatirou** [comment]
Tests not passing

**Comment by enitrat** [comment]
@obatirou need to merge ssj first

**Comment by enitrat** [comment]
all good @obatirou

File: cairo_zero/kakarot/accounts/account_contract.cairo [hunk 2]
```diff
@@ -332,6 +332,5 @@ func execute_starknet_call{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range
     to: felt, function_selector: felt, calldata_len: felt, calldata: felt*
 ) -> (retdata_len: felt, retdata: felt*, success: felt) {
     Ownable.assert_only_owner();
-    let (retdata_len, retdata) = call_contract(to, function_selector, calldata_len, calldata);
```
Comment (3docSec) [inline cairo_zero/kakarot/accounts/account_contract.cairo:335]: `call_contract` could be removed from the imports at L15

**Pull request merged**